.github/workflows/mutation.yml

```diff
@@ -55,11 +55,14 @@ jobs:
       - name: Add label to block auto merge
         if: github.event.workflow_run.event == 'pull_request' && steps.download-artifacts.outputs.found_artifact == 'true'
         run: |
+          PR_NUMBER=$(curl -H "Authorization: Bearer ${{ secrets.MUTATION_TOKEN }}" \
+            "https://api.github.com/repos/${{ github.repository }}/pulls?head=${{ github.event.workflow_run.head_repository.full_name }}:${{ github.event.workflow_run.head_branch }}" \
+            --silent | jq -r '.[0].number')
           curl \
             -X POST \
             -H "Authorization: Bearer ${{ secrets.MUTATION_TOKEN }}" \
             -H "Accept: application/vnd.github.v3+json" \
-            "https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.workflow_run.pull_requests[0].number }}/labels" \
+            "https://api.github.com/repos/${{ github.repository }}/issues/$PR_NUMBER/labels" \
```

Comment:
WDYT about using https://github.com/actions/github-script instead of curls?

Reply:
Truth be told, I just try avoid js whenever I can because of my super unfamiliarity 🤣